static/app/views/replays/detail/ai/useFetchReplaySummary.tsx

```diff
@@ -97,15 +100,26 @@ export function useFetchReplaySummary(
   // component will briefly show a completed state before the summary data query updates.
   const startSummaryRequestTime = useRef<number>(0);
 
+  // The global timeout prevents against infinite POST requests.
+  const timeoutRef = useRef<number | null>(null);
+  const [didTimeout, setDidTimeout] = useState(false);
+
   const segmentCount = replayRecord?.count_segments ?? 0;
 
   const {
     mutate: startSummaryRequestMutate,
     isError: isStartSummaryRequestError,
     isPending: isStartSummaryRequestPending,
   } = useMutation({
-    mutationFn: () =>
-      api.requestPromise(
+    mutationFn: () => {
+      if (!timeoutRef.current) {
+        timeoutRef.current = window.setTimeout(() => {
```

Comment:
probably need a ` setDidTimeout(false);` up here to handle the case when someone is retrying